File: common/src/main/java/se/casaferre/common/Temperature.java
package se.casaferre.common;

/**
 * Purpose of this class ...
 * <p>
 * Created by Jorgen Andersson (jorgen@kollektiva.se) on 2018-01-29.
 */
public class Temperature {
    /**
     * Convert Centigrades to Farenheit
     * Usage: Convert.centigradeToFahrenheit.convert(4);
     */
    public static Converter<Integer, Double> centigradeToFahrenheit = x -> new Double((x * 1.8) + 32.0);

    /**
     * Convert Farenheight to Centigrades
     * Usage: Convert.farenheightToCentigrade.convert(36);
     */
    public static Converter<Integer, Double> farenheightToCentigrade = x -> new Double((x - 32.0)  / 1.8);
}

File: common/src/main/kotlin/se/casaferre/common/Temperature.kt
package se.casaferre.common

/**
 * Purpose of this file is ...
 *
 * Created by Jorgen Andersson on 2018-06-29.
 */
object Temperature {
    /**
     * Convert Centigrades to Farenheit
     * Usage: Temperature.centigradeToFahrenheit.convert(4);
     */
    val centigradeToFahrenheit = { x: Double -> x * 1.8 + 32.0 }

    /**
     * Convert Farenheight to Centigrades
     * Usage: Temperature.farenheightToCentigrade.convert(36);
     */
    val farenheightToCentigrade = { x: Double -> (x - 32.0) / 1.8 }
}